.github/workflows/php.yml

```diff
@@ -24,25 +24,25 @@ jobs:
       with:
         php-version: ${{ matrix.php-versions }}
         coverage: xdebug
-        tools: php-cs-fixer, phpunit, phan 
-        pecl: true
+        tools: php-cs-fixer
+        extensions: ast
 
     - name: Validate composer.json and composer.lock
       run: composer validate
 
-    - name: Install dependencies
-      if: steps.composer-cache.outputs.cache-hit != 'true'
-      run: composer install --prefer-dist --no-progress --no-suggest
-            
     - name: Cache Composer packages
       id: composer-cache
       uses: actions/cache@v2
       with:
         path: vendor
-        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
```

Comment:
Since this step must be run before the composer packages installation, there is no composer.lock at this point, so I changed to composer.json (according to the docs https://github.com/shivammathur/setup-php#cache-composer-dependencies )